suite/idb-migration-utils/src/tests/parseIdbVersionString.test.ts

```diff
@@ -0,0 +1,57 @@
+import { parseIdbVersion } from '../parseIdbVersion';
+
+describe('parseIdbVersion', () => {
+    it('parses x.y.z with revision=0', () => {
+        const result = parseIdbVersion('1.2.3');
+        expect(result.semver.version).toBe('1.2.3');
+        expect(result.revision).toBe(0);
+        expect(result.versionString).toBe('1.2.3');
+    });
+
+    it('parses x.y.z.r with revision', () => {
+        const result = parseIdbVersion('25.7.0.1');
+        expect(result.semver.version).toBe('25.7.0');
+        expect(result.revision).toBe(1);
+        expect(result.versionString).toBe('25.7.0.1');
+    });
+
+    it('trims whitespace around the version', () => {
+        const result = parseIdbVersion('  2.5.3  ');
```

Comment:
For the sake of completeness, you can also test that the `.map(part => part.trim())` also works:
```suggestion
        const result = parseIdbVersion('  2 . 5 .3  ');
```